PersonApi/V1/ExceptionMiddlewareExtensions.cs

```diff
@@ -0,0 +1,86 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using PersonApi.V1.Controllers;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PersonApi.V1
+{
+    // TODO: This should go in a common NuGet package...
+
+    public static class ExceptionMiddlewareExtensions
+    {
+        public static void UseCustomExceptionHandler(this IApplicationBuilder app)
+        {
+            app.UseExceptionHandler(appError =>
+            {
+                appError.Run(async context =>
+                {
+                    await HandleExceptions(context/*, logger*/).ConfigureAwait(false);
+                });
+            });
+        }
+
+        public static async Task HandleExceptions(HttpContext context)
+        {
+            context.Response.ContentType = "application/json";
+            string message = "Internal Server Error.";
+
+            var contextFeature = context.Features.Get<IExceptionHandlerFeature>();
+            if (contextFeature != null)
+            {
+                switch (contextFeature.Error)
+                {
+                    // TODO: Handle our specific errors here.
+
+                    case Exception ex:
+                        message = ex.Message;
+                        break;
+                    default:
+                        break;
+                }
+
+                Trace.TraceError($"Request failed. {contextFeature.Error?.Message}");
+            }
+
+            var correlationId = context.Request.Headers[Constants.CorrelationId].FirstOrDefault();
+            var exceptionResult = new ExceptionResult(message, context.TraceIdentifier,
+                correlationId, context.Response.StatusCode);
+            await context.Response.WriteAsync(exceptionResult.ToString()).ConfigureAwait(false);
```

Comment:
Do we not need to call 'next' on this one at some point? (As in the next middleware in line? )

Reply:
No. Because this is the exception handler which means we've already hit an exception and need to get out.
(This is also the pattern used in the sample I copied this from - i.e. from Stacks.)